internal/events/event.go

```diff
@@ -230,21 +205,73 @@ func (e Events) hostEventsQuery(events *[]*common.Event, selectedCategories []st
 		Find(events, "cluster_id = ? AND host_id = ?", clusterID.String(), (*hostID).String())
 }
 
-func (e Events) eventsQuery(events *[]*common.Event, selectedCategories []string, clusterID *strfmt.UUID, hostID *strfmt.UUID, infraEnvID *strfmt.UUID) *gorm.DB {
+func (e Events) eventsQuery(ctx context.Context, events *[]*common.Event, selectedCategories []string, clusterID *strfmt.UUID, hostID *strfmt.UUID, infraEnvID *strfmt.UUID) *gorm.DB {
 	whereCondition := make([]string, 0)
+	user := ocm.UserNameFromContext(ctx)
+
 	if clusterID != nil {
-		whereCondition = append(whereCondition, fmt.Sprintf("cluster_id = '%s'", clusterID.String()))
+		cluster, err := common.GetClusterFromDB(e.db, *clusterID, common.UseEagerLoading)
+		if err != nil {
+			return &gorm.DB{Error: err}
+		}
+		if user != "" && !identity.IsAdmin(ctx) { // A case where there is a non-admin user in context, the service should only present resources matching to it.
+			if cluster != nil && cluster.UserName == user {
```

Comment:
per additional discussion:
1. removed the ` cluster != nil`, since it DB did not return an error we do have a cluster object.
2. keeping the `NotFound` error nested within the user comparison - since we want to return `NotFound` when:
     - cluster was not found in db
     - cluster was found in db, but the user has no access to it.